coffeecard/CoffeeCard.Models/DataTransferObjects/v2/User/WebhookUpdateUserGroupRequest.cs

```diff
@@ -0,0 +1,21 @@
+﻿using CoffeeCard.Models.Entities;
+using System.Collections.Generic;
+
+namespace CoffeeCard.Models.DataTransferObjects.v2.User
+{
+    /// <summary>
+    /// Represents a request to update user groups in bulk
+    /// </summary>
+    public class WebhookUpdateUserGroupRequest
+    {
+        /// <summary>
+        /// The hash of the request
+        /// </summary>
+        public string Hash { get; set; }
```

Comment:
After a second thought, do we need to take the hash as input? Maybe an alternative solution would be that the backend calculates the hash it self based on the list received? It would give the same effect for the backend but we relieve the client for this task.

Reply:
I agree the use case for the hash is very slim, to an extend where I'm considering to scratch it entirely? What we are really implementing with the hash is just a prevention for multiple concurrent calls spamming the database. This is probably a rare scenario regardless.

Not Modified would be the most appropriate in the case where we do not modify the database.

I believe it is better to maintain the endpoint than removing. E.g. lets say the client implementation calling the webhook is down, then it is more desirable to have an endpoint than the board editing in the database directly, thus I suggest to keep it.

You have though a good point on how the cache can limit this. I therefore either suggest to 1) remove the cache, 2) lower the cache expiration or 3) maybe pull a member list from the database and compare with the requested object. We only then update the database when the request is not equals to what is stored in the DB.
With an index on UserGroup, this would not be a heavy database operation.

Reply:
I think removing the cache completely is the best solution. If the server sends a new request there is likely a reason for it. If it is just a daily CRON-job or similar trigger action, then we should probably still make sure that the database is still "aligned". We could discuss with the board how often they want this solution to sync. If they want it based on *any* member info is updated, then we need some sort of hash I would say (consider the case where they are updating profiles just after accepting new baristas, then they will checking the list a lot). If they want it to be based on a CRON job, e.g. daily, we could either disregard the cache or instead just check "time since last request" and verify it is at least X hours old?

I quickly discussed this with Omid today, and I think we kinda thought a daily request before the café opens but after the server capacity increases could work? (probably doesn't require the server capacity to be scaled up so can also run in the middle of the night I guess).

Reply:
Lets remove the hash entirely, add the index and ship it :)